src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/connection/ConnectionManager.java

```diff
@@ -225,13 +224,13 @@ public void close() throws IllegalStateException {
 
     public void basicPublish(String exchange, String routingKey, BasicProperties props, byte[] body) throws IOException {
         Channel channel = this.channel.get();
-        waitForConnectionRecovery(channel);
+        checkConnection();
         channel.basicPublish(exchange, routingKey, props, body);
```

Comment:
Please catch exception from this method. Add blocking thread and retry if we catch exception. If on retry we exceeded count attemps, we should disable readness metric monitor which we take from method parameters .